Allow users to specify a MongoDB database name

silaev opened the pull request:
Related issue https://github.com/testcontainers/testcontainers-java/issues/2876

When one uses Testcontainer’s reuse feature, a framework and a database per a test as a way to separate(isolate) data, then specifying a database might be helpful.
For instance, in the Spring Data MongoDB a database name is bound to the MongoTemplate/
ReactiveMongoTemplate and thus it’s a tedious process to change a database name for above-mentioned testing scenario.

Given that, this PR adds ~~a withDatabaseName method for the MongoDBContainer. If the withDatabaseName mothod is not called, then a database name will default to "test" as now.~~ an overloaded getReplicaSetUrl(final String databaseName). The prior getReplicaSetUrl() remains as earlier.

bsideup commented on modules/mongodb/src/main/java/org/testcontainers/containers/MongoDBContainer.java:
```diff
@@ -20,6 +20,7 @@
     private static final int AWAIT_INIT_REPLICA_SET_ATTEMPTS = 60;
     private static final String MONGODB_VERSION_DEFAULT = "4.0.10";
     private static final String MONGODB_DATABASE_NAME_DEFAULT = "test";
+    private String databaseName = MONGODB_DATABASE_NAME_DEFAULT;
```
since it is not used anywhere but `getReplicaSetUrl`, WDYT about removing the field and adding an overloaded `getReplicaSetUrl(String)` method?

silaev replied:
It might be puzzling for users who are used to Testcontainers’ `with…` methods to create containers. However, given the rare case when a user needs to specify a MongoDB database name for testing, we can go with your solution, I’ll adjust changes.

silaev replied:
@bsideup done, please, check

rnorth commented:
Should we perhaps have a test for this, to guard against regressions?

silaev commented:
> Should we perhaps have a test for this, to guard against regressions?

Specifying a database name appeals more to frameworks. While working with a Java MongoDB driver directly, we usually call `MongoClient#getDatabase()` and that's when we specify a database name. Setting a database name in a URL does not lead to a database creation.

However in frameworks, for example in Spring Boot+Spring Data MongoDB, a database name got set to MongoTemplate during auto-configuration from a URL. Then, we don't have to call any getDatabase method at all, a database is created during a first writing call via MongoTemplate/ReactiveMongoTemplate.

Given that I've added a light-weight @DataMongoTest called MongoDBContainerDatabaseNameTest and some additional dependencies in test scope to the build.gradle file of the mongodb module.

bsideup commented:
@silaev please don't. There is no point in bringing full Spring Boot-based test just to get that `getReplicaSetUrl(String)` returns correct result. It will only add _way_ more load on CI for nothing.

silaev commented:
> @silaev please don't. There is no point in bringing full Spring Boot-based test just to get that `getReplicaSetUrl(String)` returns correct result. It will only add _way_ more load on CI for nothing.

Reverted the Spring Boot-based test, added a simple unit test shouldTestDatabaseName. 
As I mentioned before, there is no way to assert that this database is created in MongoDB physically.

silaev commented:
@rnorth @bsideup Could we merge this PR?

rnorth reviewed (approved):
👍 This looks fine to me.
